From 8397ed7416f81c7a1db594dba127dcb158088f0f Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Mon, 16 Sep 2024 15:27:12 +0200
Subject: [PATCH 1234/1695] dts: stm32: refine secure and non-secure I2C4
 pinctrl states

Explicitly state that legacy pinctrl phandles i2c4_pins_a and
i2c4_sleep_pins_a refer to non-secure I2C4 pin muxing.

Define secure I2C4 bus pinctrl states for board that use the I2C4 bus
in secure state.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 18ab8e3db1eb40c114ecb1dc25ef177a0f377051 dts: stm32: refine secure and non-secure I2C4 pinctrl states]
Change-Id: I8b481a9a7c0657137e04930c0b6c56a4d944c4e1
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/404687
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/dts/stm32mp15-pinctrl.dtsi | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/dts/stm32mp15-pinctrl.dtsi b/core/arch/arm/dts/stm32mp15-pinctrl.dtsi
index acd12c5ae..7806110a4 100644
--- a/core/arch/arm/dts/stm32mp15-pinctrl.dtsi
+++ b/core/arch/arm/dts/stm32mp15-pinctrl.dtsi
@@ -2214,6 +2214,23 @@
 	};
 
 	i2c4_pins_a: i2c4-0 {
+		pins {
+			pinmux = <STM32_PINMUX_NSEC('Z', 4, AF6)>, /* I2C4_SCL */
+				 <STM32_PINMUX_NSEC('Z', 5, AF6)>; /* I2C4_SDA */
+			bias-disable;
+			drive-open-drain;
+			slew-rate = <0>;
+		};
+	};
+
+	i2c4_sleep_pins_a: i2c4-sleep-0 {
+		pins {
+			pinmux = <STM32_PINMUX_NSEC('Z', 4, ANALOG)>, /* I2C4_SCL */
+				 <STM32_PINMUX_NSEC('Z', 5, ANALOG)>; /* I2C4_SDA */
+		};
+	};
+
+	i2c4_secure_pins_a: i2c4-sec-0 {
 		pins {
 			pinmux = <STM32_PINMUX('Z', 4, AF6)>, /* I2C4_SCL */
 				 <STM32_PINMUX('Z', 5, AF6)>; /* I2C4_SDA */
@@ -2223,7 +2240,7 @@
 		};
 	};
 
-	i2c4_sleep_pins_a: i2c4-sleep-0 {
+	i2c4_secure_sleep_pins_a: i2c4-sec-sleep-0 {
 		pins {
 			pinmux = <STM32_PINMUX('Z', 4, ANALOG)>, /* I2C4_SCL */
 				 <STM32_PINMUX('Z', 5, ANALOG)>; /* I2C4_SDA */
-- 
2.47.2

